### PR TITLE
CONTRIBUTING.md: add small blurb for first time contributors

### PR DESCRIPTION
**Pull request by lizelive**
I was confused about the maintainers file. This would have helped me.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: CONTRIBUTING.md
Comment (NobbZ) [inline CONTRIBUTING.md:56]: You do **not** need to for pass by contributions. Adding your self as a maintainer means that you want to take over long term responsibility for a "package" or "module".

Thing is, that you need to add yourself to the maintainers file in a separate commit from the changes that will reference yourself the first time.

This way, the changes to the package or module can be reverted without removing the contributor from the maintainers file, which in worst case might break other packages where they are maintainers in.

File: CONTRIBUTING.md
Comment (NobbZ) [inline CONTRIBUTING.md:60]: Please make it explicit that this are examples or do template style wording.

**Review by NobbZ — COMMENTED** [review]
Despite of my comments I think the @NixOS/documentation-team should take a look at this.

To avoid future confusion the wording really has to good.

I appreciate that something about the maintainer file gets added, as it is always hard to find.

**Comment by NobbZ** [comment]
PS: Dear docs-team, do you think it would make sense to make you a `CODEOWNERS` for this file?

File: CONTRIBUTING.md
Comment (lizelive) [reply CONTRIBUTING.md:60]: Here are the commits in an example [PR](https://github.com/NixOS/nixpkgs/pull/230958) where a user 'lizelive' updates 'starsector':

File: CONTRIBUTING.md
Comment (lizelive) [reply CONTRIBUTING.md:56]: ## Adding self as Maintainer

You only need to add yourself as a [maintainer](https://nixos.org/manual/nixpkgs/unstable/#var-meta-maintainers) if you want to take long term responsibility for a package or module. Otherwise, you can make pass by contributions without being a maintainer.

To add yourself as a maintainer, you need to make a separate commit in the maintainers file before you reference yourself in the changes to the package or module.

This way, if the changes to the package or module are reverted, you will still remain in the maintainers file and avoid breaking other packages or modules where you are a maintainer.

Comment (lizelive) [reply CONTRIBUTING.md:56]: I rephrased what you said into command, if that looks good.

**Comment by 06kellyjac** [comment]
I've added this to the project board, we can discuss it tomorrow @ the next documentation meet, thanks for CC-ing

**Review by fricklerhandwerk — COMMENTED** [review]
Looks good. Just stack commits on top, no need to discuss potential changes comments if the direction is clear. It's easier to work in actual code than pass things back and forth in comments. You can still rewrite history before merging.

A general tip on phrasing: use imperative if someone should do something. It uses fewer words and conveys the relevant information without fluff. Also try to use short sentences, roughly one per instruction. (Sometimes numbered or bulleted lists help keeping overview if each instruction needs a bit more text.)

Example: To add yourself as a maintainer, make a separate commit in the maintainers file. Then reference yourself in the changes for the package.

Another thing: I'm not sure if we should lump together contributing to Nixpkgs and NixOS. While there is a lot of overlap in many aspects, these are distinct projects with separate manuals and slightly different processes. They just happen to live in one repository because that makes a few things easier.

**Review by geri1701 — APPROVED** [review]

**Review by geri1701 — APPROVED** [review]

**Comment by lizelive** [comment]
So question before I make some those changes: I am unsure what commit message should rewrite history.

(I'm not used to repos that regularly rewrite history (I think this one recomends doing so for prs))

I would think i would add commit:

```
CONTRIBUTING.md: clarified the new contributor guide
```
or is it better to append to my commit?

Thank you.

**Comment by fricklerhandwerk** [comment]
Just append commits until approved. Maintainers can either squash directly or ask you to rewrite history if necessary. Here it should not be an issue, since the overall change will be small.

**Comment by fricklerhandwerk** [comment]
@wegank I don't see why you force pushed the branch, as @lizelive was perfectly active as a PR author. Also it made the change strictly worse, as the linked section does not really have any informative instructions to begin with. They were provided in this PR. One could argue they should go to the manual though.

@lizelive do you still want to continue? Otherwise I'd ask @wegank to fix up and follow review suggestions.

**Comment by fricklerhandwerk** [comment]
@Lizelive do you still have time to finish the PR? I think it's a great contribution, but let's make sure it strictly improves over the current situation.

I noticed that the linked section has very similar instructions. Maybe those should be moved here and linked, and that section should instead get some information on how that `lib.maintainers` is used (such as automatic notifications). The section here should instead focus on how to add yourself and what that entails.

**Comment by fricklerhandwerk** [comment]
Closing as this seems to have been implemented elsewhere already.

**Pull request closed without merging**
